### ISSUE TITLE
samples/net: The HTTP client failed to send the POST request

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#2391]
**_Reported by gurucharan p:_**

The http client is failing to send the POST request and failed with the error (-58)

Steps to replicate:
```
net-tools:
 	Terminal 1:
 	$ ./loop-socat.sh

 	Terminal 2:
 	$ sudo ./loop-slip-tap.sh

         Terminal 3:
         $ ./http-server.py

Zephyr:
 	$ cd samples/net/http_client
 	$ make pristine && make BOARD=qemu_x86 run 
```

Qemu  output:
```
 CC      arch/x86/core/fatal.o
  LD      arch/x86/core/built-in.o
  CC      arch/x86/soc/ia32/soc.o
  LD      arch/x86/soc/ia32/built-in.o
  LD      arch/x86/built-in.o
  LD      arch/built-in.o
  CC      src/main.o
  LD      src/built-in.o
  AR      libzephyr.a
  AR      libapplication.a
  LINK    zephyr.lnk
  SIDT    staticIdt.o
  GMMU    mmu_tables.o
  GDT     gdt.o
  BIN     zephyr.bin
To exit from QEMU enter: 'CTRL+a, x'
[QEMU] CPU: qemu32
qemu-system-i386: warning: Unknown firmware file in legacy mode: genroms/multiboot.bin

shell> [http-client] [INF] do_sync_http_req: HTTP server response status: OK
[http-client] [INF] do_sync_http_req: HTTP body: 161 bytes, expected: 161 bytes
[http-client] [INF] do_sync_http_req: HTTP server response status: 
[http-client] [INF] do_sync_http_req: HTTP server response status: OK
[http-client] [ERR] do_sync_http_req: Error detected during HTTP msg processing
[http-client] [ERR] do_sync_http_req: Cannot send POST request (-58)
[http-client] [INF] main: Done!
```

(Imported from Jira ZEP-2561)

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Andrei Laperie_ , please set the priority of this bug.  Thx.

**Comment by nashif** [comment]
**_by Jukka Rissanen:_**

The commit 8ebaf299 ("net: http: dont timeout on HTTP requests w/o body") broke the http_client and caused this issue. If I revert the commit, then the http-client starts to work properly again. I have no yet a fix to this issue.

**Comment by nashif** [comment]
**_by Jukka Rissanen:_**

I am assigning this temporarily to _Michael Scott_  if he has any suggestion how to solve this issue.

**Comment by nashif** [comment]
**_by Jukka Rissanen:_**

Some detailed information about this:

The error in samples/net/http_client/src/main.c happens when we do synchronous HTTP request. The async methods seem to work just fine.

**Comment by nashif** [comment]
**_by Michael Scott:_**

I bisected it to my commit from last week:
https://github.com/zephyrproject-rtos/zephyr/commit/8ebaf29927c29df50488fdd9ef5ea6c8b47c9323
"net: http: dont timeout on HTTP requests w/o body"

I'm taking a look at it now.

**Comment by nashif** [comment]
**_by Michael Scott:_**

Debugging a new patch to fix that now.

**Comment by nashif** [comment]
**_by Michael Scott:_**

Pushed this PR to fix this issue: https://github.com/zephyrproject-rtos/zephyr/pull/1227

NOTE: I tested http_client and a private sample which was how the original issue surfaced.  And both are now happy.

**Comment by nashif** [comment]
**_by Jukka Rissanen:_**

Thanks Mike, I appreciate your support!

**Comment by nashif** [comment]
**_by Chandrakala Kempanna:_**

_gurucharan p_ Please Verify

**Comment by nashif** [comment]
**_by gurucharan p:_**

It seems the issue has been fixed with v1.9.0-rc2. So closing the Jira.
